README.md
AltoWithCSS3
============
Le configurateur permet de choisir d'afficher ou non le menu (simpleMenu ou Menu requis) et quelle largeur de page vous voulez (880px, 1024px ou 1280px).

Dimensions pour les images de bannière :
- affichage 880 : 723px x 150px
- affichage 1024 : 866px x 150px
- affichage 1280 : 1220px x 150px

Si vous voulez en ajouter d'autres, il est impératif de se conformer aux règles suivantes :
- les nommer de manière identique "roundX.ext" où X est un chiffre
- en cas d'ajout (et non de remplacement), il faut modifier :
  * le fichier tpl/user_head.html et indiquer en ligne 3 (```var round = parseInt(Math.random()*3);```) le nombre total d'image de bannière (3 par défaut)
  * le fichier css correspondant à la largeur de page choisie (css/880.css, css/1024.css ou css/1280.css) en prenant exemple sur la déclaration css : ```.round0 { background : transparent url(../img/1024/round0.jpg) no-repeat top left; }```


changelog
v2.0.2 - 11/01/2023 Pierre Van Glabeke
* modifs css

v2.0.1 - 07/01/2023 Pierre Van Glabeke
* bug configurateur

v2.0 - 19/12/2022 Pierre Van Glabeke
* ajout largeur 1280

v1.9.1 - 12/12/2022 JC Denis
* update pour dc2.24

v1.9 - 26/11/2019 Pierre Van Glabeke
* modifs html
* dc2.15 requis

v1.8.1 - 16/04/2019 Pierre Van Glabeke
* suppression code postcount dans page.html

v1.8 - 31/10/2017 Pierre Van Glabeke
* modifs html
* modifs img

v1.7 - 30/03/2016 Pierre Van Glabeke
* modifs css (page et css3)
* modif appel plugin Menu
* dc2.9 requis

v1.6.1 - 12/03/2016 Pierre Van Glabeke
* modifs css

v1.6 - 01/03/2016 Pierre Van Glabeke
* commentaire dans footer
* modifs css et banner

v1.5.1 - 09/01/2016 Pierre Van Glabeke
* modifs css

v1.5 - 08/01/2016 Pierre Van Glabeke
* modifs css
* ajout localisations (de, es)
* héritage thème mustek
* dc2.8 requis

v1.4.2 - 18/03/2015 Pierre Van Glabeke
* suppression balises obsolètes html5
* suppression mention menuFreshy (configurateur)

v1.4.1 - 06/03/2015 Pierre Van Glabeke
* marqueur version dans user_head.html
* modifs css
* modifs post et page (coquille attachments)
* modif _footer.html
* correction setting

v1.4 - 16/12/2014 Pierre Van Glabeke
* html5 + ajouts css

v1.3 - 21/07/2014 Pierre Van Glabeke
* modifs css + localisation
* ajout combo menu : menu (avec sous-menus) ou simplemenu ou aucun

v1.2 - 09/04/2014 Pierre Van Glabeke
* modifs css + localisation + lien dans footer

v1.1 - 04/03/2014 Pierre Van Glabeke
* modifs menu dans _config et _public + css

v1.0 - 21/02/2014 Pierre Van Glabeke
* Reprise du thème altowithjQuery et passage à CSS3 pour les arrondis.

_public.php
<?php
/**
 * @brief AltoWithCSS3, a theme for Dotclear 2
 *
 * @package Dotclear
 * @subpackage Theme
 *
 * @author Pierre Van Glabeke
 * @copyright http://creativecommons.org/licenses/by-nc/2.0/fr/
 */

if (!defined('DC_RC_PATH')) {
    return;
}

l10n::set(__DIR__ . '/locales/' . dcCore::app()->lang . '/main');

# appel css menu
dcCore::app()->addBehavior('publicHeadContent', 'altowithcss3_publicHeadContent');

function altowithcss3_publicHeadContent()
{
    // menu
    $style = dcCore::app()->blog->settings->themes->altowithcss3_menu;
    if (!preg_match('/^menu|simplemenu|menuno$/', (string) $style)) {
        $style = 'simplemenu';
    }

    $url = dcCore::app()->blog->settings->system->themes_url . '/' . dcCore::app()->blog->settings->system->theme;
    echo '<link rel="stylesheet" type="text/css" media="screen" href="' . $url . '/css/' . $style . ".css\" />\n";

    // width
    $style = dcCore::app()->blog->settings->themes->altowithcss3_width;
    if (!preg_match('/^880|1024|1280$/', (string) $style)) {
        $style = '1024';
    }

    $url = dcCore::app()->blog->settings->system->themes_url . '/' . dcCore::app()->blog->settings->system->theme;
    echo '<link rel="stylesheet" type="text/css" media="screen" href="' . $url . '/css/' . $style . ".css\" />\n";
}


menu.AltoWithCSS3.php
<?php
/**
 * @brief AltoWithCSS3, a theme for Dotclear 2
 *
 * @package Dotclear
 * @subpackage Theme
 *
 * @author Pierre Van Glabeke
 * @copyright http://creativecommons.org/licenses/by-nc/2.0/fr/
 */

# Template style menu
$template_theme_style = [
    'LI_ACTIVE'              => 'class="current_page_item"',
    'LI_LAST'                => 'class="last_menu"',
    'LI_ACTIVE_LAST'         => 'class="current_page_item last_menu"',
    'LI_PARENT'              => 'class="top_parent"',
    'LI_FIRST_PARENT'        => 'class="first_menu top_parent"',
    'LI_LAST_PARENT'         => 'class="last_menu top_parent"',
    'LI_ACTIVE_FIRST_PARENT' => 'class="first_menu top_parent"',
    'LI_ACTIVE_LAST_PARENT'  => 'class="last_menu top_parent"',
    'LI_SMPARENT'            => 'class="current_page_parent"',
    'LI_ACTIVE_SM'           => 'class="current_page_item"',
];

# Widget style menu
$widget_theme_style = [
    'W_LI_ACTIVE'    => 'class="current_page_item"',
    'W_LI_ACTIVE_SM' => 'class="current_page_item"',
];


_config.php
<?php
/**
 * @brief AltoWithCSS3, a theme for Dotclear 2
 *
 * @package Dotclear
 * @subpackage Theme
 *
 * @author Pierre Van Glabeke
 * @copyright http://creativecommons.org/licenses/by-nc/2.0/fr/
 */

if (!defined('DC_CONTEXT_ADMIN')) {
    return;
}

//PARAMS

# Translations
l10n::set(__DIR__ . '/locales/' . dcCore::app()->lang . '/main');

# Default values
$default_menu  = 'simplemenu';
$default_width = '1024';

# Settings
$my_menu  = dcCore::app()->blog->settings->themes->altowithcss3_menu;
$my_width = dcCore::app()->blog->settings->themes->altowithcss3_width;

# Menu type
$altowithcss3_menu_combo = [
    __('simpleMenu') => 'simplemenu',
    __('Menu')       => 'menu',
    __('none')       => 'menuno',
];

# Width type
$altowithcss3_width_combo = [
    __('880')  => '880',
    __('1024') => '1024',
    __('1280') => '1280',
];

// POST ACTIONS

if (!empty($_POST)) {
    try {
        dcCore::app()->blog->settings->addNamespace('themes');

        # Menu type
        if (!empty($_POST['altowithcss3_menu']) && in_array($_POST['altowithcss3_menu'], $altowithcss3_menu_combo)) {
            $my_menu = $_POST['altowithcss3_menu'];
        } elseif (empty($_POST['altowithcss3_menu'])) {
            $my_menu = $default_menu;
        }
        dcCore::app()->blog->settings->themes->put('altowithcss3_menu', $my_menu, 'string', 'Menu to display', true);

        # Width type
        if (!empty($_POST['altowithcss3_width']) && in_array($_POST['altowithcss3_width'], $altowithcss3_width_combo)) {
            $my_width = $_POST['altowithcss3_width'];
        } elseif (empty($_POST['altowithcss3_width'])) {
            $my_width = $default_width;
        }
        dcCore::app()->blog->settings->themes->put('altowithcss3_width', $my_width, 'string', 'Width to display', true);

        // Blog refresh
        dcCore::app()->blog->triggerBlog();

        // Template cache reset
        dcCore::app()->emptyTemplatesCache();

        dcPage::success(__('Theme configuration has been successfully updated.'), true, true);
    } catch (Exception $e) {
        dcCore::app()->error->add($e->getMessage());
    }
}

// DISPLAY

# Menu
echo
'<div class="fieldset"><h4>' . __('Customizations') . '</h4>' .
'<p class="field"><label>' . __('Menu to display:') . '</label>' .
form::combo('altowithcss3_menu', $altowithcss3_menu_combo, $my_menu) .
'</p>' .
'<p class="info">' . __('Plugins menu allowed: <a href="http://plugins.dotaddict.org/dc2/details/menu">Menu</a> plugin or simpleMenu.') . '</p>';

# Width type
echo
'<p class="field"><label>' . __('Display width:') . '</label>' .
form::combo('altowithcss3_width', $altowithcss3_width_combo, $my_width) .
'</p>' .
'</div>';
